### PR TITLE
Recipes-index

### LINKED ISSUE #4: [4pt] Implement Recipes list view and controller actions
- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.

### PR DESCRIPTION
**Pull request by eamador1**
In this project, the following has been done:

Should display a list of recipes created by the logged-in user as in the wireframe
Should lead to recipe details

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[ show edit update destroy ]
```
Comment (Estete9) [inline app/controllers/recipes_controller.rb]: Great application for before_action! 
We can improve it by removing the edit and update.

File: app/controllers/recipes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  end
+
+  # GET /recipes/new
+  def new
+    @recipe = Recipe.new
+  end
+
+  # GET /recipes/1/edit
+  def edit
+  end
```
Comment (Estete9) [inline app/controllers/recipes_controller.rb]: Great work with the actions for recipes. The requirements ask to implement all actions except for update. Please remove the edit action.

File: app/controllers/recipes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    respond_to do |format|
+      if @recipe.update(recipe_params)
+        format.html { redirect_to recipe_url(@recipe), notice: "Recipe was successfully updated." }
+        format.json { render :show, status: :ok, location: @recipe }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
```
Comment (Estete9) [inline app/controllers/recipes_controller.rb]: Great work with the actions for recipes. The requirements ask to implement all actions except for update. Please remove the update action.

File: app/views/recipes/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  <div>
+    <%= form.label :public, style: "display: block" %>
+    <%= form.check_box :public %>
+  </div>
+
+  <div>
+    <%= form.label :user, style: "display: block" %>
+    <%= form.number_field :user %>
+  </div>
+
```
Comment (Estete9) [inline app/views/recipes/_form.html.erb]: This form is looking great, let's just remove this div as we don't need to specify the user

File: app/views/recipes/_recipe.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<div id="<%= dom_id recipe %>">
+  <h2>
+    <%= recipe.name %>
+  </h2>
+  <div>
+    <% if can? :destroy, recipe %>
+    <%= link_to "REMOVE", recipe_path(recipe.id), data: { confirm: 'Are you sure?', turbo_method: :delete }, turbo_method: :delete %>
+  </div>
```
Comment (Estete9) [inline app/views/recipes/_recipe.html.erb:9]: This is looking great, just add `<% end %>` between the lines 7 and 8

File: app/views/recipes/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<div id="recipes">
+  <% @recipes.each do |recipe| %>
+    <%= render recipe %>
+  <% end %>
```
Comment (Estete9) [inline app/views/recipes/index.html.erb]: This works almost perfect! Let's add a lin_to so we can open each recipe in a new page (recipes#show)
Ex:
`<% @recipes.each do |recipe| %>`
`  <%= link_to recipe_path(recipe.id) do %>`
 `    <%= render recipe %>`
`  <% end %>`
`<% end %>`

File: config/routes.rb [hunk 7]
```diff
@@ -1,8 +1,9 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  resources :inventories, except: [:update, :edit]
 
+  resources :recipes
```
Comment (Estete9) [inline config/routes.rb]: Great work with the routes, it's almost perfect. Please modify this so the available routes are only for the actions that we are using. You can decide to exclude the ones we are not going to use (like the routes for inventories below in the code), or you can manually add the routes like this `resources :recipes, only: [:index, :new, :create, (etc.)]`

**Review by Estete9 — CHANGES_REQUESTED** [review]
Hi @eamador1 
I'll review this PR. ✋



Good job so far it looks quite good!
There are some recommendations I would like you to take into account.

Highlights
 You have written a clean code ✅
 The distribution, padding, and margins are perfect. ✅
 Good PR title/description ✅
Required Changes ♻️
 Good job adding the elements in the list in the main section, but we can improve it a little bit more together. 💪 
 
Check the comments under the review.

Cheers and Happy coding!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Estete9 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/Estete9">
   @Estete9
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/esteban-palacios-5030a772/">
   linkedin-esteban-palacios-5030a772
</a>

</td>
<td>

<a href="https://twitter.com/NaughTban">
   @NaughTban
</a>

</td>
</tr>

**Comment by eamador1** [comment]
Since the last code review, the following was done:
----

Add authorization
Update controller actions
Update routes

**Review by Estete9 — COMMENTED** [review]
## Hi @eamador1 , 

Your feature is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Great use of modules ✅ 
- Linters are green ✅ 
- Meaningful **Github** commits ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Estete9 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/Estete9">
   @Estete9
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/esteban-palacios-5030a772/">
   linkedin-esteban-palacios-5030a772
</a>

</td>
<td>

<a href="https://twitter.com/NaughTban">
   @NaughTban
</a>

</td>
</tr>
</table>

**Review by Estete9 — APPROVED** [review]
## Hi @eamador1 , 

Your feature is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Great use of modules ✅ 
- Linters are green ✅ 
- Meaningful **Github** commits ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Estete9 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/Estete9">
   @Estete9
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/esteban-palacios-5030a772/">
   linkedin-esteban-palacios-5030a772
</a>

</td>
<td>

<a href="https://twitter.com/NaughTban">
   @NaughTban
</a>

</td>
</tr>
</table>

**Pull request merged**